optuna/visualization/matplotlib/_pareto_front.py

```diff
@@ -65,22 +70,100 @@ def objective(trial):
             A list of indices indicating the axis order. If :obj:`None` is specified,
             default order is used.
 
+            .. warning::
+                Deprecated in v3.0.0. This feature will be removed in the future. The removal of
+                this feature is currently scheduled for v5.0.0, but this schedule is subject to
+                change. See https://github.com/optuna/optuna/releases/tag/v3.0.0.
+        targets:
+            A function that returns a tuple of target values to display.
+            The argument to this function is :class:`~optuna.trial.FrozenTrial`.
+
+            .. note::
+                Added in v3.0.0 as an experimental feature. The interface may change in newer
+                versions without prior notice.
+                See https://github.com/optuna/optuna/releases/tag/v3.0.0.
+
     Returns:
         A :class:`matplotlib.axes.Axes` object.
 
     Raises:
         :exc:`ValueError`:
-            If the number of objectives of ``study`` isn't 2 or 3.
+            If ``targets`` is ``None`` when your objective studies have more than 3 objectives.
+        :exc:`ValueError`:
+            If ``targets`` returns something other than sequence.
+        :exc:`ValueError`:
+            If the number of target values to display isn't 2 or 3.
+        :exc:`ValueError`:
+            If ``targets`` is specified for empty studies and ``target_names`` is ``None``
+        :exc:`ValueError`:
+            if using both ``targets`` and ``axis_order``.
     """
 
     _imports.check()
+    if axis_order is not None:
+        warnings.warn(
+            "`axis_order` has been deprecated in v3.0.0. "
+            "This feature will be removed in v5.0.0. "
+            "See https://github.com/optuna/optuna/releases/tag/v3.0.0.",
+            DeprecationWarning,
+        )
+    trials = _get_trials(study, include_dominated_trials)
+    if targets is not None and axis_order is not None:
+        raise ValueError(
+            "Using both ``targets`` and ``axis_order`` is not supported. "
```

Comment:
This part is for console, not documentation so I would like to make the same suggestion as  https://github.com/optuna/optuna/pull/3341#discussion_r815405293

```suggestion
            "Using both `targets` and `axis_order` is not supported. "
```